### ISSUE TITLE
Circusd not cleaning up children started from shell scripts

### ISSUE THREAD

**Issue by remko** [circus-tent/circus#1077]
I have a Java process started (without `exec`) from a Bash script, with the following setup:

    [watcher:foo]
    cmd = /path/to/script
    copy_env = True
    stop_children = True

When I interrupt Circusd, it says:

    2018-08-26 08:25:45 circus[50748] [INFO] Arbiter exiting
    2018-08-26 08:25:45 circus[50748] [INFO] foo stopped

However, the Java process is still running. When directly invoking the Java executable, it is shutdown properly. The shell script isn't under my control, though. 

I have similar issues with other servers started from shell scripts without `exec`. Is there a way to ensure that these are cleaned up? I'm currently using another supervisor that doesn't leave any of these processes behind.

I'm using Circus 0.15.0 on Mac OS X 10.13.6.

**Comment by k4nar** [comment]
Indeed, I think there is something wrong with the option `stop_children` as reported here: https://github.com/circus-tent/circus/issues/1044.

One good solution, on Linux at least, would be to use cgroups. But so far no one has invested enough time I guess.
